src/Storages/ObjectStorage/StorageObjectStorageCluster.cpp

```diff
@@ -153,7 +160,7 @@ void StorageObjectStorageCluster::updateQueryForDistributedEngineIfNeeded(ASTPtr
             queryToString(query));
     }
 
-    configuration->setFunctionArgs(arguments->children);
+    configuration->getTableFunctionArguments(arguments->children);
```

Comment:
In https://github.com/Altinity/ClickHouse/pull/615#discussion_r1949607293, I suggested the arguments to be created inside `getTableFunctionArguments` instead of populating a parameter. Any specific reason you did it this way?

If there is one, I am fine with this approach

Reply:
Just to avoid copying after function call. `children` is a vector, not a pointer.

Reply:
Changes not in `arguments` but in `arguments->children`
`children` has type `ASTs`
https://github.com/ClickHouse/ClickHouse/blob/master/src/Parsers/IAST.h#L37
It's a variant of vector, not a single pointer
https://github.com/ClickHouse/ClickHouse/blob/master/src/Parsers/IAST_fwd.h#L14

Reply:
As discussed on slack, let's make the function return it as suggested in the original comment